server/src/main/java/io/unitycatalog/server/service/TableService.java

```diff
@@ -2,55 +2,168 @@
+      #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #schema, OWNER, USE_SCHEMA)
+      """)
+  @AuthorizeKey(METASTORE)
+  public HttpResponse createTable(
+      @AuthorizeKeys({
+            @AuthorizeKey(value = SCHEMA, key = "schema_name"),
+            @AuthorizeKey(value = CATALOG, key = "catalog_name")
+          })
+          CreateTable createTable) {
     assert createTable != null;
-    TableInfo createTableResponse = TABLE_REPOSITORY.createTable(createTable);
-    return HttpResponse.ofJson(createTableResponse);
+    TableInfo tableInfo = TABLE_REPOSITORY.createTable(createTable);
+    initializeAuthorizations(tableInfo);
+    return HttpResponse.ofJson(tableInfo);
   }
 
   @Get("/{full_name}")
-  public HttpResponse getTable(@Param("full_name") String fullName) {
+  @AuthorizeExpression("""
+          #authorize(#principal, #metastore, OWNER) ||
+          (#authorizeAny(#principal, #schema, OWNER, USE_SCHEMA) && #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #table, OWNER, SELECT))
+          """)
+  @AuthorizeKey(METASTORE)
+  public HttpResponse getTable(@Param("full_name") @AuthorizeKey(TABLE) String fullName) {
     assert fullName != null;
     TableInfo tableInfo = TABLE_REPOSITORY.getTable(fullName);
     return HttpResponse.ofJson(tableInfo);
   }
 
   @Get("")
+  @AuthorizeExpression("#defer")
   public HttpResponse listTables(
       @Param("catalog_name") String catalogName,
       @Param("schema_name") String schemaName,
       @Param("max_results") Optional<Integer> maxResults,
       @Param("page_token") Optional<String> pageToken,
       @Param("omit_properties") Optional<Boolean> omitProperties,
       @Param("omit_columns") Optional<Boolean> omitColumns) {
-    return HttpResponse.ofJson(
-        TABLE_REPOSITORY.listTables(
+
+    ListTablesResponse listTablesResponse = TABLE_REPOSITORY.listTables(
             catalogName,
             schemaName,
             maxResults,
             pageToken,
             omitProperties.orElse(false),
-            omitColumns.orElse(false)));
+            omitColumns.orElse(false));
+
+    filterTables("""
+            #authorize(#principal, #metastore, OWNER) ||
+            (#authorizeAll(#principal, #schema, OWNER, USE_SCHEMA) && #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG)) ||
+            (#authorize(#principal, #schema, USE_SCHEMA) && #authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #table, OWNER, SELECT))
+            """, listTablesResponse.getTables());
+
+    return HttpResponse.ofJson(listTablesResponse);
   }
 
   @Delete("/{full_name}")
-  public HttpResponse deleteTable(@Param("full_name") String fullName) {
+  @AuthorizeExpression("""
+          (#authorizeAny(#principal, #catalog, OWNER, USE_CATALOG) && #authorizeAny(#principal, #schema, OWNER, USE_SCHEMA) && #authorize(#principal, #table, OWNER))
```

Comment:
Principals that can delete a table
- metastore owner
- catalog owner
- schema owner with USE_CATALOG
- table owner with USE_CATALOG and USE_SCHEMA